proposals/0002-root-signature-in-clang.md

```diff
@@ -936,6 +936,199 @@ Operands:
+  -T rootsig_1_0 -E RS /Fo RS.bin
+  -T rootsig_1_1 -E RS /Fo RS.bin
+```
+
+Behaviour:
+
+ - Parse and perform syntactic validations of "entry" root signature
+ - Perform the 'basic' sub-set of validations that are available without access
+to resource bindings
+ - Produces a DXIL container with only the serialized "entry" root signature
+as the only part
+
+ - Note: It is not possible to use `-rootsig-define` to overwrite which root
+signature will be used as the "entry" root signature.
+
+Design Notes: Commonly used. Planned in Clang.
+
+#### Option `-Qstrip-rootsignature`
+
+Omits the root signature part (RTS0) from the produced DXIL Container.
+
+Usage:
+
+```
+  -Qstrip-rootsignature
+```
+
+Behaviour:
+
+ - Parse and perform syntactic validations of "entry" root signature
+ - Perform the 'basic' sub-set of validations that are available without access
+to resource bindings
+ - Produces the DXIL container with the RTS0 omitted
+
+Design Notes: Usable within .hlsl source files. Planned in Clang unless there is
+a signal of it being unused.
+
+Design Notes: Plan to match DXC behaviour directly such that we still do
+the parsing and basic validations of the root signataure.
+
+#### Option `-setrootsignature`
+
+Overwrite the root signature part (RTS0) with the RTS0 of provided DXIL
+Container.
+
+Usage:
+
+```
+  -setrootsignature <DXIL Container>
+```
+
+Behaviour:
+
+ - Insert/overwrite the RTS0 part of the compiled DXIL Container with the RST0
+contents of the input container
+ - Does not perform any validation with RTS0 of the provided DXIL Container
+ - Requires input to be a valid DXIL container with or without an RST0 part
+ - Note: Even when provided, this will parse/validate a root signature
+specified as an attribute (or another above) before dropping it.
+
+Design Notes: Planned in Clang. If compiling a stand-alone DXIL Container with
+just the RTS0 is useful then this is presumably also required in that workflow.
+
+#### Option `-extractrootsignature`
+
+Specifies to compile the shader as normal but will only output the RTS0 part
+into the produced DXIL Container.
+
+Usage:
+
+```
+  -extractrootsignature /Fo <DXIL Container>
+```
+
+Behaviour:
+
+ - Behaviour is the same as the
+[root signature target option](#target-root-signature-version), with the
+exception that it will run all validations with the resource bindings in the
+shader code
+
+Design Notes: Not planned in Clang. This is on the assumption that running the
+additional validations doesn't provide additional utility over the root
+signature target option and compiling the shader regularly.
+
+#### Option `-verifyrootsignature`
+
+Given a shader defined from a DXIL Container, perform validations with the
+root signature part (RTS0) of the other DXIL Container
+
+Usage:
+
+```
+  -dumpbin <Shader DXIL Container> -verifyrootsignatures <Root Signature DXIL Container>
+```
+
+Note: `dumpbin` is required to load a binary instead of performing compilation
```

Comment:
This is great, but I wonder if we can't just have a `-dumpbin` section we refer to from each of the other options that can apply to a loaded DXIL container using this option.